Samples/blazor-add-in/outlook-blazor-add-in/outlook-blazor-add-in/Pages/Index.razor.js

```diff
@@ -0,0 +1,177 @@
+                    }
+
+                    if (fileExtension !== "") {
+                        console.log(`Index.razor.js(getEmailData): Processing file extension '${fileExtension}'...`);
+                        console.log("Attachment Value: " + returnedAttachments[i].value.content);
+
+                        //await item.getAttachmentContentAsync(attachmentID, handleAttachmentsCallback);
+
+                        return {
+                            AttachmentId: item.attachments[i].id,
+                            AttachmentName: attachmentName,
+                            Subject: item.subject,
+                            AttachmentType: attachmentType, 
+                            Inline: fileinline,
+                            AttachmentBase64Data: returnedAttachments[i].value.content
+                        };
+                    } else {
+                        console.error(`Index.razor.js(getEmailData): Could not parse file extension for ${attachmentName}`);
+                        continue;
+                    }
+                }
+            }
+            catch (e) {
+                console.log("----------------------------------------------- never executed", e);
+            }
+        }
+        else {
+            return {
+                AttachmentId: "",
+                AttachmentName: "No Attachments",
+                Subject: "",
+                AttachmentType: "",
+                Inline: false,
+                AttachmentBase64Data: ""
+            };
+        }
+    } catch (err) {
+        console.error(`Index.razor.js(getEmailData) Catch Exception: ${err}`);
+        subject = `${err}`;
+        return { Subject: subject };
+    }
+}
+
+function getFileExtension(fileName) {
+    var a = fileName.split(".");
+    if (a.length === 1 || (a[0] === "" && a.length === 2)) {
+        return "";
+    }
+    return a.pop();
+}
+
+function getAttachments(item) {
+
+    // getAttachmentContentAsync() is only supported above 1.8
+    if (!Office.context.requirements.isSetSupported("Mailbox", "1.8")) {
+        return [];
+    }
+
+    const attachments = item.attachments;
+
+    // If you need to filter, uncomment...
+    // -----------------------------------
+    //    = item.attachments.filter(
+    //    (attachment) => attachment.attachmentType === "file" && attachment.isInline === true
+    //);
+
+    console.log("Filtered attachments:", attachments.length);
+
+    if (!attachments) {
+        return [];
+    }
+
+    return Promise.all(
+        attachments.map(
+            (attachment) =>
+                new Promise((resolve) => {
+                    console.log(`Index.razor.js (getAttachments) attachment ID returned: ` + attachment.id);
+                    item.getAttachmentContentAsync(attachment.id, (result) => resolve(result));
+                })
+        )
+    );
+}
+
+//    function handleAttachmentsCallback(result) {
```

Comment:
Can remove commented code no longer used.

Reply:
True, the current code however is using Promises and wanted to see if we can do the same with a 'more modern' async await option. Couldn't get that to work properly though (doesn't seem to await properly when used in a Callback).
It can be removed though as it is now part of the check-in history.

Reply:
Done ...